src/platform/plugins/shared/navigation/public/top_nav_menu/top_nav_menu_item.tsx

```diff
@@ -53,7 +53,13 @@ export function TopNavMenuItem(props: TopNavMenuItemProps) {
 
   function handleClick(event: MouseEvent<HTMLButtonElement | HTMLAnchorElement>) {
     if (isDisabled()) return;
-    if (props.href && isModifiedEvent(event)) return;
+    if (props.href) {
+      if (isModifiedEvent(event)) {
+        return;
+      }
+
+      event.preventDefault();
+    }
```

Comment:
This one was tricky. Tests starting failing in this backport because the Discover "New" button started refreshing the page. I don't know enough about `history` to know exactly why, but it seems like we didn't hit this before because the `href` path was the same as the pushed path. In this PR I changed to a redirect URL so the path changed. I guess `preventDefault` isn't needed when the paths match, but now it is. We didn't hit it in 9.1 because of #211802.